cmd/kubectl-gadget/bcck8s.go

```diff
@@ -240,7 +250,7 @@ func bccCmd(subCommand, bccScript string) func(*cobra.Command, []string) {
 
 		// tcptop only works on one pod at a time
 		if subCommand == "tcptop" {
-			if nodeParam == "" || podnameParam == "" {
+			if nodeParam == "" || podnameParam == "" || jsonOutput {
 				contextLogger.Fatalf("tcptop only works with --node and --podname")
```

Comment:
I think we need a different error message for `jsonOutput==true`.

Reply:
I updated it.